From: Ben Hutchings <benh@debian.org>
Date: Tue, 25 Aug 2020 23:27:40 +0100
Subject: tools/include/uapi: Fix <asm/errno.h>

tools/include/uapi/asm/errno.h currently attempts to include
non-existent arch-specific headers for ia64 and xtensa.  Remove
these cases so that <asm-generic/errno.h> is used instead.

It does not use the arch-specific header for parisc, so add a
case for that.

References: https://buildd.debian.org/status/fetch.php?pkg=linux&arch=ia64&ver=5.8.3-1%7Eexp1&stamp=1598340829&raw=1
Signed-off-by: Ben Hutchings <benh@debian.org>
[Salvatore Bonaccorso: Inclusion for ia64 was dropped in upstream 5.10.32,
refresh patch for context changes]
Signed-off-by: Salvatore Bonaccorso <carnil@debian.org>
---
--- a/tools/include/uapi/asm/errno.h
+++ b/tools/include/uapi/asm/errno.h
@@ -9,8 +9,8 @@
 #include "../../../arch/alpha/include/uapi/asm/errno.h"
 #elif defined(__mips__)
 #include "../../../arch/mips/include/uapi/asm/errno.h"
-#elif defined(__xtensa__)
-#include "../../../arch/xtensa/include/uapi/asm/errno.h"
+#elif defined(__hppa__)
+#include "../../../arch/parisc/include/uapi/asm/errno.h"
 #else
 #include <asm-generic/errno.h>
 #endif
